### ISSUE TITLE
`Style/AutoResourceCleanup` shouldn't flag `File.open(…).close`.

### ISSUE THREAD

**Issue by reitermarkus** [bbatsov/rubocop#4444]
The `Style/AutoResourceCleanup` cop should not flag the following (`touch` + `chmod`):

```ruby
File.open('file', 'w', 0o777).close
```

--------

## Expected behavior

Given a file with the content above, `rubocop --only Style/AutoResourceCleanup` should output:

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Actual behavior

Given a file with the content above, `rubocop --only Style/AutoResourceCleanup` outputs:

```
Inspecting 1 file
C

Offenses:

open.rb:1:1: C: Use the block version of File.open.
File.open(file, 'w', 0o100777).close
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.0.0 universal.x86_64-darwin16)
```
